internal/thelma/clients/vault/masking_round_tripper_test.go

```diff
@@ -0,0 +1,31 @@
+package vault
+
+import (
+	vaulthelper "github.com/broadinstitute/thelma/internal/thelma/clients/vault/testing"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func Test_MaskingRoundTripper(t *testing.T) {
+	fakeVaultServer := vaulthelper.NewFakeVaultServer(t)
+	fakeVaultServer.SetSecret("secret/foo/bar", map[string]interface{}{"abc": "xyz"})
+
+	client := fakeVaultServer.Server().Client()
+
+	maskFnCalled := false
+
+	maskingTransport := MaskingRoundTripper{
+		inner: client.Transport,
+		maskFn: func(secrets ...string) {
+			assert.Len(t, secrets, 1)
+			assert.Equal(t, "xyz", secrets[0])
+			maskFnCalled = true
+		},
+	}
+	client.Transport = maskingTransport
+
+	_, err := client.Get(fakeVaultServer.Server().URL + "/v1/secret/foo/bar")
+	require.NoError(t, err)
+	assert.True(t, maskFnCalled)
```

Comment:
Is there anyway to assert the secret is actually being masked in output, not just that the masking function has been called?

Reply:
So, the actual masking logic lives in the logging package and there are [test cases](https://github.com/broadinstitute/thelma/blob/main/internal/thelma/app/logging/logging_test.go#L142) there that make sure that it works as expected. All the round tripper does is delegate to the `logging.MaskSecret` function.

Reply:
The logging package is like the only package in Thelma that has global state (passing around logging objects sucks) so reliably integration testing it here could actually be pretty tough.